src/v/datalake/record_schema_resolver.cc

```diff
@@ -404,48 +396,45 @@ latest_protobuf_schema_resolver::resolve_buf_type(
         co_return schema_res.error();
     }
     auto shared_schema = schema_res.value();
-    auto pb_def_res = shared_schema->visit(ss::make_visitor(
-      [](const ppsr::protobuf_schema_definition& pb_def)
-        -> checked<ppsr::protobuf_schema_definition, type_resolver::errc> {
-          return pb_def;
+    auto resolve_res = shared_schema->visit(ss::make_visitor(
+      [this, &latest_schema, &shared_schema](
+        const ppsr::protobuf_schema_definition& pb_def)
+        -> checked<resolved_type, type_resolver::errc> {
+          std::vector<int32_t> offsets;
+          if (const auto& explicit_name = protobuf_message_name_) {
+              auto res = compute_message_offsets(pb_def, *explicit_name);
+              if (res.has_error()) {
+                  return res.error();
+              }
+              offsets = std::move(res.value());
+          } else {
+              offsets = {0};
+          }
+          return translate_protobuf_schema(
+            pb_def, latest_schema.id, offsets, std::move(shared_schema));
       },
-      [](const auto&)
-        -> checked<ppsr::protobuf_schema_definition, type_resolver::errc> {
+      [&latest_schema, &shared_schema](const ppsr::avro_schema_definition& def)
+        -> checked<resolved_type, type_resolver::errc> {
+          return translate_avro_schema(
+            def, latest_schema.id, std::move(shared_schema));
```

Comment:
Oh Avro support is surprising from the CL, but great! I think the release note needs an update.

Reply:
>which allows one to use Iceberg topics with protobuf and not having to encode the messages using the Schema Registry wire format.

Specifically this part that calls out protobuf specifically, given it's true for Avro too now

Reply:
I actually don't want to encourage avro usage at all, because it fundamentally does not allow schema evolution. Not supporting it explicitly seems a bit harsh and @mattschumpert was for allowing it. Schema evolution in avro works by having the new and old schemas and evolving the parsed data, in this mode we'll fail to parse old avro records. I will make sure we have a note on the docs this is not recommended to use avro with this mode, but having the mode based on how we lookup the schema is better than something specific for protobuf and is easier to explain.

Reply:
Yea fair. It does seem like a decent addition regardless, especially if users know there won't be schema changes. It'd be interesting to understand the set of customers using Avro without the SR format to see if there is some other mode we ought to be supporting

Reply:
Yeah agreed. I think this no one does this (schema registry was initially created only for avro and protobuf then json came later). If they do they have to be communicating the schema out of band somehow (maybe via a header?) and I am interested to know if there are existing patterns out there.